libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -58,6 +58,10 @@ class WiFiClientRxBuffer {
         {
             if(!_buffer){
                 _buffer = (uint8_t *)malloc(_size);
+                if(!_buffer) {
```

Comment:
Please add an error log like `log_e("Not enough memory to allocate buffer");`

this helps a lot on debug =)